atomic_reactor/plugins/post_generate_maven_metadata.py

```diff
@@ -37,98 +32,15 @@ def __init__(self, workflow):
-
-            for build_archive in build_archives:
-                checksum_type = koji.CHECKSUM_TYPES[build_archive['checksum_type']]
-                components.append({
-                    'type': 'kojifile',
-                    'filename': build_archive['filename'],
-                    'filesize': build_archive['size'],
-                    'checksum': build_archive['checksum'],
-                    'checksum_type': checksum_type,
-                    'nvr': nvr_request.nvr,
-                    'archive_id': build_archive['id'],
-                })
-
-        return components
-
-    def get_pnc_build_metadata(self, pnc_requests):
-        builds = pnc_requests.get('builds', [])
-
-        if not builds:
-            return {}
-
-        pnc_build_metadata = {'builds': []}
-
-        for build in builds:
-            pnc_build_metadata['builds'].append({'id': build['build_id']})
-
-        return pnc_build_metadata
-
-    def process_url_requests(self, url_requests):
-        download_queue = []
-
-        for url_request in url_requests:
-            artifact_checksums = {algo: url_request[algo] for algo in
-                                  hashlib.algorithms_guaranteed
-                                  if algo in url_request}
-
-            artifact = {
-                'url': url_request['url'],
-                'checksums': artifact_checksums,
-                'filename': os.path.basename(url_request['url'])
-            }
-
-            if 'source-url' not in url_request:
-                self.no_source_artifacts.append(artifact)
-                msg = f"No source-url found for {url_request['url']}.\n"
-                self.log.warning(msg)
-                msg += 'fetch-artifacts-url without source-url is deprecated\n'
-                msg += 'to fix this please provide the source-url according to ' \
-                       'https://osbs.readthedocs.io/en/latest/users.html#fetch-artifacts-url-yaml'
-                self.log.user_warning(msg)
-                continue
-
-            source_url = url_request['source-url']
-
-            checksums = {algo: url_request[('source-' + algo)] for algo in
-                         hashlib.algorithms_guaranteed
-                         if ('source-' + algo) in url_request}
-
-            if source_url not in self.source_url_to_artifacts:
-                self.source_url_to_artifacts[source_url] = [artifact]
-                # source_url will mostly be gerrit URLs that don't have filename
-                #  in the URL itself so we'll have to get filename from URL response
-                target = os.path.basename(source_url)
-
-                download_queue.append(DownloadRequest(source_url, target, checksums))
-            else:
-                self.source_url_to_artifacts[source_url].append(artifact)
-        return download_queue
-
-    def download_sources(self, download_queue):
+    def get_remote_source_files(self, download_queue):
```

Comment:
It would be good to add type annotations here, this code currently expects that `download_queue` is a list of `DownloadRequest` (but that may have to change due to the requirement of plugin results being JSON-compatible)

Plus we have adding type annotations in the checklist now :)

Reply:
added type annotations